### PR TITLE
Add email feature test

### PR DESCRIPTION
**Pull request by RoadBytes**
First Half of Week 5 course 3 Tea Leaf Academy

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class PagesController < ApplicationController
```
Comment (RoadBytes) [inline app/controllers/pages_controller.rb]: I just made a generic pages controller for the invalid token response.  Is this a kosher way to go about handling invalid tokens?

File: config/environments/production.rb [hunk 2]
```diff
@@ -19,12 +19,14 @@
...
+
+  ActionMailer::Base.smtp_settings = {
+    :port           => ENV['MAILGUN_SMTP_PORT'],
+    :address        => ENV['MAILGUN_SMTP_SERVER'],
+    :user_name      => ENV['MAILGUN_SMTP_LOGIN'],
+    :password       => ENV['MAILGUN_SMTP_PASSWORD'],
+    :domain         => 'yourapp.heroku.com',
+    :authentication => :plain,
+  }
+  ActionMailer::Base.delivery_method = :smtp
```
Comment (RoadBytes) [inline config/environments/production.rb]: Mailgun is pretty cool.  By the way, is `config.action_mailer.smtp_settings` and `ActionMailer::Base.smtp_settings` the same?

File: app/controllers/pages_controller.rb (hunk 1 above)
Comment (tomtomecek) [inline app/controllers/pages_controller.rb]: Its fine, counts as a static page to me as well.

File: config/environments/production.rb (hunk 2 above)
Comment (tomtomecek) [inline config/environments/production.rb]: I think, they should be the same.

File: app/controllers/invitations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  def create
+    @invitation = Invitation.new(strong_invitation_params.merge!(inviter_id: current_user.id))
+    @invitation.generate_token
+
+    if @invitation.save
+      AppMailer.send_invitation(@invitation).deliver
+      flash[:success] = "Invitation was sent!"
+      redirect_to new_invitation_path
+    else
+      flash[:danger] = "Invitation was NOT sent"
```
Comment (tomtomecek) [inline app/controllers/invitations_controller.rb]: Its `flash.now` when you render after.

HOMEWORK: Find out why its better to use `now` method with flash on render.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -27,4 +42,15 @@ def create
...
   end
+
+  def handle_invitation token
+    invitation = Invitation.find_by(token: token)
+
+    if invitation
+      inviter = User.find_by id: invitation.inviter_id
+      @user.follow   inviter
+      inviter.follow @user
+      invitation.update_attributes(token: nil)
```
Comment (tomtomecek) [inline app/controllers/users_controller.rb]: You could have a `remove_token` method here and this logic pushed to the Model. Also `update` is the same as `update_attributes` but shorter :)

File: app/models/user.rb [hunk 5]
```diff
@@ -36,7 +37,10 @@ def can_follow?(user)
     !(self == user || self.is_following?(user))
   end
 
-  def generate_token
-    self.token = SecureRandom.urlsafe_base64
+  def follow leader
```
Comment (tomtomecek) [inline app/models/user.rb]: I would definitely do not omit brackets when defining method.

File: app/views/invitations/new.html.haml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+%section.invite.container
+  .row
+    .col-sm-10.col-sm-offset-1
+      = bootstrap_form_for @invitation, class: "invite" do |f|
+        %header
+          %h1 Invite a friend to join MyFlix!
+        %fieldset
+          = f.text_field :recipient_name, label: "Friend's Name"
+          = f.text_field :recipient_email, label: "Friend's Email"
```
Comment (tomtomecek) [inline app/views/invitations/new.html.haml]: `email_field` yields more benefits, when you expect the email.

File: spec/fabricators/invitation_fabricator.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+require 'faker'
```
Comment (tomtomecek) [inline spec/fabricators/invitation_fabricator.rb]: I don't think you need to require it here.

File: spec/models/user_spec.rb [hunk 8]
```diff
@@ -68,4 +68,19 @@
...
+
+  describe "#follow" do
+    it "follows another user" do
+      alice = Fabricate(:user)
+      joe   = Fabricate(:user)
+      joe.follow alice
+      expect(joe.reload.is_following? alice).to eq true
+    end
+
+    it "it does not follow one's self" do
```
Comment (tomtomecek) [inline spec/models/user_spec.rb]: minus `it`

File: app/controllers/invitations_controller.rb (hunk 3 above)
Comment (RoadBytes) [inline app/controllers/invitations_controller.rb]: flash is by default available on the next request so add `flash.now` for a render so it's available on the same request...  ie, no redirect = same request.

> By default, adding values to the flash will make them available to the next request, but sometimes you may want to access those values in the same request. For example, if the create action fails to save a resource and you render the new template directly, that's not going to result in a new request, but you may still want to display a message using the flash. To do this, you can use flash.now in the same way you use the normal flash:

Comment (tomtomecek) [inline app/controllers/invitations_controller.rb]: Yes, and if you do just `flash` your error message will pop on next users GET request, so its better to use `.now`.

**Pull request merged**